hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -940,11 +940,17 @@ public AbfsRestOperation read(final String path, final long position, final byte
         abfsUriQueryBuilder, cachedSasToken);
 
     URL url = createRequestUrl(path, abfsUriQueryBuilder.toString());
-    if (url.toString().contains(WASB_DNS_PREFIX)) {
-      url = changePrefixFromBlobtoDfs(url);
+    AbfsRestOperationType opType;
+    if (getAbfsConfiguration().getPrefixMode() == PrefixMode.BLOB) {
```

Comment:
Should add a fallback for read API like others and have raised a PR for unified fallback, condition should be config controlled

Reply:
Added